impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -379,6 +379,22 @@ std::optional<Rect> GaussianBlurFilterContents::GetFilterCoverage(
   return expanded_source_coverage.TransformBounds(entity.GetTransform());
 }
 
+namespace {
+Vector2 ExtractScale(const Matrix& matrix) {
+  Vector2 entity_scale_x = matrix * Vector2(1.0, 0.0);
```

Comment:
Matrix::Decompose here?

Reply:
I looked at that and it was way more calculations for things we just throw away.  It's doing things like calculating the determinant.  It returns an optional result probably for this reason.  This one is guaranteed to have a result too.

Reply:
These matrix mults are just recomputing the basis vectors, which could just be pulled out of the matrix instead: `return Vector2(matrix.GetBasisX().Length(), matrix.GetBasisY().Length())`

`Matrix::GetScale` does the same thing, but includes the Z basis and returns a `Vector3`. Adding a `Matrix::GetScaleXY` seems like a good idea since we're doing this in a few places now.